### ISSUE TITLE
render.* doesn't type check it's color argument in most cases

### ISSUE THREAD

**Issue by Kefta** [Facepunch/garrysmod-issues#3768]
``render.DrawSphere(vector_origin, 16, 50, 50, NULL)`` and ``render.DrawBox(vector_origin, angle_zero, Vector(-32, -32, -32), Vector(32, 32, 32), NULL)`` will interpret the incorrectly-typed color argument as Color(255, 255, 255) instead of throwing an error.

**Comment by robotboy655** [comment]
This can probably be said about any optional argument.

**Comment by thegrb93** [comment]
Wouldn't it also hit performance?

**Comment by Kefta** [comment]
Other optional arguments check their types - these are odd ones out. Could you at least throw non-halting errors so people can know if they're using the args incorrectly?

**Comment by robotboy655** [comment]
What "other optional arguments"? This behavior should affect most if not all functions with optional arguments. Can you name one that handles this as you think it should be handled?

**Comment by Kefta** [comment]
You're right, only vanilla Lua funcs type check their optional arguments. So.. any plans for Garry's Mod 14?
